### DISCUSSION TITLE
Allow adding pathfinder goals to an entity (vanilla and custom)

### DISCUSSION THREAD

**Discussion by kangarko** [PaperMC/Paper#11462] [Ideas]
### Is your feature request related to a problem?

N/A

### Describe the solution you'd like.

Extending the https://github.com/PaperMC/Paper/issues/2617 to be able for developers to add custom navigation to entities like we do in the NMS class without the need to use NMS.

### Describe alternatives you've considered.

Using Citizens plugin. This is good enough but comes with limitations and each entity needs to be wrapped as NPC.

**Comment by Nacioszeczek** [comment]
https://jd.papermc.io/paper/1.21.1/org/bukkit/entity/Mob.html#getPathfinder()
